foundry/pixi/containers/canvasLayers/controlsLayer.d.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * A CanvasLayer for displaying UI controls which are overlayed on top of other layers.
+ *
+ * We track three types of events:
+ * 1) Cursor movement
+ * 2) Ruler measurement
+ * 3) Map pings
+ */
+declare class ControlsLayer extends CanvasLayer {
+  constructor();
+
+  /**
+   * Cursor position indicators
+   */
+  cursors: PIXI.Container | null;
+
+  /**
+   * A mapping of user IDs to Cursor instances for quick access
+   */
+  protected _cursors: Record<string, Cursor>;
+
+  /**
+   * Door control icons
+   */
+  doors: PIXI.Container | null;
+
+  /**
+   * Status effect icons
+   */
+  effects: unknown | null;
+
+  /**
+   * Ruler tools, one per connected user
+   */
+  rulers: PIXI.Container | null;
+
+  /**
+   * A convenience mapping of user IDs to Ruler instances for quick access
+   */
+  protected _rulers: Record<string, Ruler>;
+
+  /**
+   * Canvas selection rectangle
+   */
+  select: PIXI.Graphics | null;
+
+  // The controls layer is always interactive
+  interactiveChildren: true;
+
+  /** @override */
+  static get layerOptions(): CanvasLayer.LayerOptions;
+
+  /**
+   * A convenience accessor to the Ruler for the active game user
+   */
+  get ruler(): Ruler | null;
```

Comment:
Just asking in general, because I have been doing it myself more and more: If the return type directly depends on another function, wouldn't it be better to reference that function directly? In this case it would be
```typescript
ReturnType<ControlsLayer['getRulerForUser']>
```

Reply:
I can answer my own question after some experimentation: It just makes stuff more rigid and harder to use. So we should rather not do it.

Reply:
Actually it seems I need to revise my second statement again. `tsc` just sent me on a wild goose chase, making me think those are a problem, when the problem was actually somewhere completely different.